pkg/terminal/colorize/colorize.go

```diff
@@ -78,20 +79,34 @@ func Print(out io.Writer, path string, reader io.Reader, startLine, endLine, arr
 		toks = append(toks, colorTok{tok, int(start), int(end)})
 	}
 
-	emit(token.PACKAGE, f.Package, token.NoPos)
-
 	for _, cgrp := range f.Comments {
 		for _, cmnt := range cgrp.List {
 			emit(token.COMMENT, cmnt.Pos(), cmnt.End())
 		}
 	}
 
-	ast.Inspect(f, func(n ast.Node) bool {
-		if n == nil {
+	ast.Inspect(f, func(node ast.Node) bool {
+		if node == nil {
 			return true
 		}
 
-		switch n := n.(type) {
+		switch n := node.(type) {
+		case *ast.File:
+			emit(token.PACKAGE, f.Package, token.NoPos)
+			return true
+		case *ast.FuncType:
+			// this implies `node` is a method of interface
```

Comment:
I may misunderstand `ast.FuncType.Func==token.NoPos`, before I thought only a method of interface could satisfy this condition.

Here we should highlight the keyword `func` ranther than the function name. So when `n.Pos() == token.NoPos` we just return, otherwise `emits(token.Func, n.Pos, token.NoPos)` to hightlight the keyword `func`.